client/allocrunner/consul_hook.go

```diff
@@ -197,24 +198,26 @@ func (h *consulHook) prepareConsulTokensForServices(services []*structs.Service,
 		}
 
 		if err := h.getConsulTokens(clusterName, service.Identity.Name, tokens, req); err != nil {
-			mErr.Errors = append(mErr.Errors, err)
+			mErr = multierror.Append(mErr, err)
 			continue
 		}
 	}
 
-	return mErr.ErrorOrNil()
+	return mErr
 }
 
-func (h *consulHook) getConsulTokens(cluster, identityName string, tokens map[string]map[string]*consulapi.ACLToken, req map[string]consul.JWTLoginRequest) error {
+func (h *consulHook) getConsulTokens(cluster, identityName string, tokens map[string]map[string]*consulapi.ACLToken, req map[string]consul.JWTLoginRequest) *multierror.Error {
 	client, err := h.clientForCluster(cluster)
 	if err != nil {
-		return err
+		return &multierror.Error{Errors: []error{err}}
 	}
 
 	// get consul acl tokens
 	t, err := client.DeriveTokenWithJWT(req)
 	if err != nil {
-		return err
+		// DeriveTokenWithJWT actually returns a multierror, so we assert the
+		// type to make error formatting from the hook level more readable.
+		return err.(*multierror.Error)
```

Comment:
Not sure what happened, but this diff doesn't include all my changes from https://github.com/hashicorp/nomad/commit/16d6cfaad00d43ced253a3862a2f46c65fb985cd. 

This is particular is the one that I'm worried about.

Reply:
ugh, must've been some git merge fail, apologies. I fix.

Reply:
actually it seems that 16d6cfa and your branch in general have been merged completely, they simply don't have the changes you want. I'll put them in manually.

Reply:
I think [1da57ed](https://github.com/hashicorp/nomad/pull/19133/commits/1da57edb68491ac3849fee1628b502c1a658fe3e) provides a good fix.